spread-shellcheck

```diff
@@ -161,7 +164,7 @@ def checkfile(path):
 
     if path.endswith('spread.yaml') and 'suites' in data:
         # check suites
-        with ThreadPoolExecutor() as executor:
+        with ThreadPoolExecutor(max_workers=N) as executor:
```

Comment:
I think it would be more correct to use a single thread pool executor, then we share the real number of threads, not run one pool from another.

Reply:
Done in https://github.com/snapcore/snapd/pull/9493 -- closing this branch.